core/evalir.ml

```diff
@@ -138,20 +138,20 @@ struct
 
   (** {0 Evaluation} *)
   let rec value env : Ir.value -> Value.t = function
-    | `Constant (Constant.Bool   b) -> `Bool b
-    | `Constant (Constant.Int    n) -> `Int n
-    | `Constant (Constant.Char   c) -> `Char c
-    | `Constant (Constant.String s) -> Value.box_string s
-    | `Constant (Constant.Float  f) -> `Float f
-    | `Variable var -> lookup_var var env
+    | Constant (Constant.Bool   b) -> `Bool b
```

Comment:
`Value.t` is another candidate for conversion to a normal variant. But not in this patch, I think.

Reply:
Indeed! However, it is actually used as a polymorphic variant in value.ml and will require some additional work in order to become a regular datatype.